src/storage/composed.js

```
@@ -47,14 +56,21 @@ const ComposedStorage = async (storage1, storage2) => {
    * @instance
    */
   const get = async (hash) => {
-    let value = await storage1.get(hash)
-    if (!value) {
-      value = await storage2.get(hash)
-      if (value) {
-        await storage1.put(hash, value)
+    try {
+      let value = await storage1.get(hash)
+      if (!value) {
+        value = await storage2.get(hash)
+        if (value) {
+          await storage1.put(hash, value)
+        }
       }
+      return value
+    } catch (e) {
+      console.error(`Error while getting hash ${hash} from composed storage`, e)
+      events.emit('error', e)
     }
-    return value
+
+    return undefined
```

I'm concerned that this changes the expected behavior of the function by not re-emitting the error.

---

Ah so like if an update error occurs we wouldn't emit the get error?

---

Yeah, suppose that the sync protocol depends on the error being emitted when a block can't be gotten to abort the stream, but we suppress the error and then the stream continues erroneously. On the other hand, what you did here seems to match the behavior in the included `level.js`, so I'm going to approve.